service/handlers.go

```diff
@@ -0,0 +1,152 @@
+package service
+
+import (
+	"encoding/json"
+	"github.com/bar41234/bar_book_service/models"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+const (
+	ErrorMsgIdNotFound         = "Error: Id was not found!"
+	ErrorMsgInvalidPutRequest  = "Error: Invalid PUT request"
+	ErrorMsgInvalidPostRequest = "Error: Invalid POST request"
+)
+
+func Ping(c *gin.Context) {
+	username := c.Query("username")
+	CacheAdd(username, c)
+
+	c.JSON(http.StatusOK, map[string]string{
+		"PING!": "PONG!",
```

Comment:
not comment to use special charters as a JSON key: you can replace it with 
```suggestion
		"message": "Ping! PONG!",
```